### ISSUE TITLE
tasks not picked up for app when app uses AppConfig

### ISSUE THREAD

**Issue by ghukill** [arteria/django-background-tasks#130]
Perhaps this is obvious to others, but this took a little investigating to figure out why tasks were not picked up.

We had been using the [`AppConfig`](https://docs.djangoproject.com/en/2.0/ref/applications/#django.apps.AppConfig) to explicitly configure an app.

e.g.
```
class CoreConfig(AppConfig):
	
	name = 'core'

	def ready(self):

		# do things here...
```

And our `INSTALLED_APPS` looked like this:
```
INSTALLED_APPS = [
    'core.apps.CoreConfig',
    'django.contrib.admin',
    ...
    'django_extensions',
    'background_task',    
]
```

This was not picking up any tasks from `core.tasks`.  But if we move away from using `AppConfig`, which we didn't need anymore, and changed our `INSTALLED_APPS` to the following, things worked as expected:

```
INSTALLED_APPS = [
    'core',
    'django.contrib.admin',
    ...
    'django_extensions',
    'background_task',    
]
```

I don't know if the `autodiscover()` method could account for this, or if it just belongs in the documentation somewhere (which is relatively sparse with regards to `tasks.py`, itself something I had to gleam from other closes issues), but thought I would share here.

That said, thanks for a great plugin!

**Comment by joncard1** [comment]
Yes, this seems to be because of the line in autodiscover that says:

    for app in settings.INSTALLED_APPS:
        try:
            app_path = import_module(app).__path__
        except (AttributeError, ImportError):
            continue

The line "import_module(app)" does not work with a value of "core.apps.CoreConfig"; it will say "ModuleNotFoundError: No module named 'core.apps.CoreConfig'; 'core.apps' is not a package". It will work with "core", then search for "core/tasks.py" and find the tasks, which will allow it to schedule tasks in the database that are named in that file. Came here to report that issue, after having spent some time trying to find it.

**Comment by ruben-mv** [comment]
Same problem here!

As a workarround to can use django-backgorund-tasks and AppConfig together, you can use
`default_app_config = 'core.apps.CoreConfig'`
on your app `__init__.py`, and use 
```
INSTALLED_APPS = [
    'core',
    ...
]
```
on `settings.py`.

With this method, `autodiscover()` will work, and django will use your AppConfig.

**Comment by JnBrymn** [comment]
I ran across this issue tonight.

According to django docs (here https://docs.djangoproject.com/en/3.0/ref/settings/#std:setting-INSTALLED_APPS), the strings listed in INSTALLED_APPS should be "an application configuration class (preferred), or a package containing an application." The django "preferred" isn't supported because of the problem described above.

This old issue is also caused by the same problem: https://github.com/arteria/django-background-tasks/issues/34

**Comment by pixyj** [comment]
I found another workaround by importing the `tasks` module in the `AppConfig`'s `ready` method

```
class CoreConfig(AppConfig):
    name = 'hire.core'

    def ready(self):
        from . import tasks  # noqa
```
